src/fondant/pipeline.py

```diff
@@ -66,7 +70,20 @@ def __init__(
         )
         self.name = self.component_spec.name.replace(" ", "_").lower()
         self.cache = self._configure_caching_from_image_tag(cache)
-        self.arguments = self._set_arguments(arguments)
+        self.cluster_type = cluster_type
+        self.client_kwargs = client_kwargs
+
+        self.arguments = self._set_arguments(arguments)  #
+
+        self._add_component_argument(
+            "input_partition_rows",
+            input_partition_rows,
+            validate_partition_number,
+        )
+        self._add_component_argument("cache", cache)
+        self._add_component_argument("cluster_type", cluster_type)
+        self._add_component_argument("client_kwargs", client_kwargs)
```

Comment:
For the first time reading the code, it wasn't clear from the initial perspective where the ComponentOp arguments are defined. The magic is actually hidden inside the `_set_arguments` function. I would propose a slightly different method, such as `_add_component_argument`. In my opinion, this would make it a bit clearer to understand what's happening just by looking at the class constructor.
@PhilippeMoussalli wdyt?

Reply:
I think it makes sense indeed, looks more clear